libs/components/indicators/testing/src/icon/icon-harness.spec.ts

```diff
@@ -0,0 +1,234 @@
+  );
+}
+
+async function validateIconSize(
+  iconHarness: SkyIconHarness,
+  fixture: ComponentFixture<TestComponent>,
+  iconSize: string | undefined
+): Promise<void> {
+  fixture.componentInstance.size = iconSize;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getIconSize()).toBeResolvedTo(iconSize);
+}
+
+async function validateVariant(
+  iconHarness: SkyIconHarness,
+  fixture: ComponentFixture<TestComponent>,
+  variant: string | undefined
+): Promise<void> {
+  fixture.componentInstance.variant = variant;
+  fixture.detectChanges();
+  await expectAsync(iconHarness.getVariant()).toBeResolvedTo(
+    variant ? variant : ''
+  );
+}
+
+const iconTypes = ['fa', 'skyux'];
+const variants = ['line', 'solid'];
+const sizes = [
+  '2xs',
+  'xs',
+  'sm',
+  'lg',
+  '2xl',
+  '1x',
+  '2x',
+  '3x',
+  '4x',
+  '5x',
+  '6x',
+  '7x',
+  '8x',
+  '9x',
+  '10x',
+];
+
+describe('Icon harness', () => {
+  async function setupTest(options: { dataSkyId?: string } = {}): Promise<{
+    iconHarness: SkyIconHarness;
+    fixture: ComponentFixture<TestComponent>;
+    loader: HarnessLoader;
+    pageLoader: HarnessLoader;
+  }> {
+    await TestBed.configureTestingModule({
+      declarations: [TestComponent],
+      imports: [SkyIconModule],
+    }).compileComponents();
+
+    const fixture = TestBed.createComponent(TestComponent);
+    const loader = TestbedHarnessEnvironment.loader(fixture);
+    const pageLoader = TestbedHarnessEnvironment.documentRootLoader(fixture);
+
+    const iconHarness: SkyIconHarness = options.dataSkyId
+      ? await loader.getHarness(
+          SkyIconHarness.with({ dataSkyId: options.dataSkyId })
+        )
+      : await loader.getHarness(SkyIconHarness);
+    fixture.detectChanges();
+
+    return { iconHarness, fixture, loader, pageLoader };
+  }
+
+  it('should setup the icon test harness', async () => {
+    const { iconHarness, fixture } = await setupTest();
+
+    await validateIconName(
+      iconHarness,
+      fixture,
+      fixture.componentInstance.iconName
+    );
+    await validateFixedWidth(
+      iconHarness,
+      fixture,
+      fixture.componentInstance.fixedWidth
+    );
+    await validateIconType(
+      iconHarness,
+      fixture,
+      fixture.componentInstance.iconType
+    );
+    await validateIconSize(
+      iconHarness,
+      fixture,
+      fixture.componentInstance.size
+    );
+    await validateVariant(
+      iconHarness,
+      fixture,
+      fixture.componentInstance.variant
+    );
+  });
+
+  it('should return the correct icon name for all icon types and variants', async () => {
+    const { iconHarness, fixture } = await setupTest();
+    for (const type of iconTypes) {
+      fixture.componentInstance.iconType = type;
+      for (const variant of variants) {
+        fixture.componentInstance.variant = variant;
+        await validateIconName(
+          iconHarness,
+          fixture,
+          fixture.componentInstance.iconName
```

Comment:
yup! i was also trying to future proof in case we add more `iconTypes` or `variants`. 
same for future proofing of the variable name, so if for some reason the `house` icon name goes away and this unit test fails, we only change it in one location. But i can change it for readability.